kedro-datasets/kedro_datasets/netcdf/netcdf_dataset.py

```diff
@@ -0,0 +1,155 @@
+                files.
+            temppath: Local temporary directory, used when reading from remote storage,
+                since NetCDF files cannot be directly read from remote storage.
+            load_args: Additional options for loading NetCDF file(s).
+                Here you can find all available arguments when reading single file:
+                https://xarray.pydata.org/en/stable/generated/xarray.open_dataset.html
+                Here you can find all available arguments when reading multiple files:
+                https://xarray.pydata.org/en/stable/generated/xarray.open_mfdataset.html
+                All defaults are preserved.
+            save_args: Additional saving options for saving NetCDF file(s).
+                Here you can find all available arguments:
+                https://xarray.pydata.org/en/stable/generated/xarray.Dataset.to_netcdf.html
+                All defaults are preserved.
+            fs_args: Extra arguments to pass into underlying filesystem class
+                constructor (e.g. `{"cache_regions": "us-east-1"}` for
+                ``s3fs.S3FileSystem``).
+            credentials: Credentials required to get access to the underlying filesystem.
+                E.g. for ``GCSFileSystem`` it should look like `{"token": None}`.
+        """
+        self._fs_args = deepcopy(fs_args) or {}
+        self._credentials = deepcopy(credentials) or {}
+        protocol, path = get_protocol_and_path(filepath)
+        if protocol == "file":
+            self._fs_args.setdefault("auto_mkdir", True)
+        else:
+            if temppath is None:
+                raise ValueError(
+                    "Need to set temppath in catalog if NetCDF file exists on remote "
+                    + "filesystem"
+                )
+            self._temppath = Path(temppath)
+        self._protocol = protocol
+        self._filepath = PurePosixPath(path)
+
+        self._storage_options = {**self._credentials, **self._fs_args}
+        self._fs = fsspec.filesystem(self._protocol, **self._storage_options)
+
+        # Handle default load and save arguments
+        self._load_args = deepcopy(self.DEFAULT_LOAD_ARGS)
+        if load_args is not None:
+            self._load_args.update(load_args)
+        self._save_args = deepcopy(self.DEFAULT_SAVE_ARGS)
+        if save_args is not None:
+            self._save_args.update(save_args)
+
+    def _load(self) -> xr.Dataset:
+        load_path = get_filepath_str(self._filepath, self._protocol)
+
+        is_multifile = True if "*" in str(load_path) else False
+
+        # If NetCDF(s) are on any type of remote storage, need to sync to local to open.
+        # Kerchunk could be implemented here in the future for direct remote reading.
+        if self._protocol != "file":
+            log.info("Syncing remote NetCDF file to local storage.")
+
+            # `get_filepath_str` drops remote protocol prefix.
+            load_path = self._protocol + "://" + load_path
```

Comment:
Tested with AWS and GCS, but not sure if this is generalized enough. `get_filepath_str` consistently drops the prefix of URI's from object storage, which is problematic for the subsequent `fs.get`

Reply:
Yeah filepath mangling in fsspec is tricky... we have a related open issue about this https://github.com/kedro-org/kedro/issues/3196